### PR TITLE
add TypedResponseListener to streams

### PR DESCRIPTION
**Pull request by rwiskerke**
If this PR is acceptable, I'll also supply a PR backporting this change to 2.3.x.

### CODE REVIEW

File: elastic4s-streams/src/main/scala/com/sksamuel/elastic4s/streams/BulkIndexingSubscriber.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ import scala.util.{Failure, Success}
  */
 class BulkIndexingSubscriber[T] private[streams](client: ElasticClient,
                                                  builder: RequestBuilder[T],
-                                                 config: SubscriberConfig)
+                                                 config: TypedSubscriberConfig[T])
```
Comment (sksamuel) [inline elastic4s-streams/src/main/scala/com/sksamuel/elastic4s/streams/BulkIndexingSubscriber.scala]: This change means existing code will not compile, so I think we should add an apply method to do the conversion into a TypedSubscriberConfig[Nothing].

Comment (rwiskerke) [inline elastic4s-streams/src/main/scala/com/sksamuel/elastic4s/streams/BulkIndexingSubscriber.scala]: This constructor is private to the `streams` package, so it shouldn't break any existing code of library users, right?

**Comment by sksamuel** [comment]
Ah I missed that.

**Comment by sksamuel** [comment]
Merged thanks.

**Comment by rwiskerke** [comment]
If you create a 2.3 branch I will submit a PR for that as well. I already did the backport, but I don't think I can submit it without a branch which I can target?

**Comment by rwiskerke** [comment]
I based my forks branch on commit 604cd06

**Pull request merged**